dbfit-java/core/src/test/java/dbfit/environment/TypeSpecifierTest.java

```diff
@@ -0,0 +1,35 @@
+package dbfit.environment;
+
+import org.junit.*;
+import static org.junit.Assert.*;
+
+import java.sql.SQLException;
+import java.util.Map;
+
+import dbfit.api.DBEnvironment;
+
+import dbfit.util.TypeSpecifier;
+
+public class TypeSpecifierTest {
+
+    private class DummyTypeSpecifier implements TypeSpecifier {
+        public Object specify(Object o) throws SQLException {
+            return new String("abcdef");
+        }
+    }
+
+    @Test
+    public void castTypeTest() {
+        DBEnvironment env = dbfit.api.DbEnvironmentFactory.newEnvironmentInstance("Dummy");
+        env.setTypeSpecifier(java.util.Date.class, new DummyTypeSpecifier());
+        Map<Class<?>, TypeSpecifier> tsm = env.getTypeSpecifierMap();
+        TypeSpecifier ts = tsm.get(java.util.Date.class);
+        String s = null;
+        try {
+            s = (String) ts.specify(new java.util.Date());
```

Comment:
> excuse my delayed and not so clear posts

Of course, no problem, please excuse my vast PRs! And thanks for the explanations above too.

> it may be just creating plain StatementExecution initialized with specific parameters.

I think that's kind of where I started off with PR #432.I'd rather not have to return that that PR now!

> What would happen if we pass not exactly java.util.Date but e.g. a subclass of it?

I will fail to transform the input object as it won't find a `TypeSpecifier`. I've deliberately kept things simple for this PR and didn't create functionality similar to what's in `TypeNormaliser`.